### ISSUE TITLE
grumble softmax抽出器を噛ませる

### ISSUE THREAD

**Issue by shunyooo** [shunyooo/Sentence-VAE#3]
![Group 81](https://user-images.githubusercontent.com/17490886/69956678-fb388980-1543-11ea-86ac-b18b8c31f1e7.png)

**Comment by shunyooo** [comment]
このモデルだめ
grumble softmax の出力がただの中間層になってしまっている。
grumble softmax の出力が単語idを表すように Embedding層を加える。

![Group 83](https://user-images.githubusercontent.com/17490886/70036036-524b6680-15f7-11ea-8efa-58cb15f8bd7c.png)

懸念点
- Embedding層はあらかじめ学習したものを使うか
- softmaxとEmbeddingを掛け合わせた行列がちゃんと複数の単語を示すようになるか

**Comment by shunyooo** [comment]
Embedding検証
![image](https://user-images.githubusercontent.com/17490886/70040609-2b912e00-15ff-11ea-913b-59d1cf4b5be8.png)

- `model.embedding(index)` は `torch.matmul(onehot, model.embedding.weight)` で再現できる
- 複数indexを入れた時、sumになる

**Comment by shunyooo** [comment]
パッケージあり
https://pytorch.org/docs/stable/nn.functional.html?highlight=gumbel#torch.nn.functional.gumbel_softmax
